syncer/ddl_test.go

```diff
@@ -207,149 +213,144 @@ func (s *testSyncerSuite) TestResolveDDLSQL(c *C) {
 	}
 
 	for i, sql := range sqls {
-		result, err := syncer.parseDDLSQL(sql, p, "test")
+		qec := &queryEventContext{
+			eventContext: ec,
+			ddlSchema:    "test",
+			originSQL:    sql,
+			appliedDDLs:  make([]string, 0),
+			p:            parser.New(),
+		}
+		stmt, err := syncer.parseDDLSQL(qec)
 		c.Assert(err, IsNil)
-		c.Assert(result.ignore, IsFalse)
-		c.Assert(result.isDDL, IsTrue)
+		_, ok := stmt.(ast.DDLNode)
+		c.Assert(ok, IsTrue)
 
-		statements, _, err := syncer.splitAndFilterDDL(ec, p, result.stmt, "test")
+		qec.splitedDDLs, err = parserpkg.SplitDDL(stmt, qec.ddlSchema)
+		c.Assert(err, IsNil)
+		// c.Assert(targetSQLs[i], HasLen, len(qec.splitedDDLs))
+		err = syncer.preprocessDDL(qec)
 		c.Assert(err, IsNil)
-		c.Assert(statements, DeepEquals, expectedSQLs[i])
-		c.Assert(targetSQLs[i], HasLen, len(statements))
+		c.Assert(qec.appliedDDLs, DeepEquals, expectedSQLs[i])
+		c.Assert(targetSQLs[i], HasLen, len(qec.appliedDDLs))
 
-		for j, statement := range statements {
-			s, _, _, err := syncer.routeDDL(p, "test", statement)
+		for j, statement := range qec.appliedDDLs {
+			s, _, _, err := syncer.routeDDL(qec, statement)
 			c.Assert(err, IsNil)
 			c.Assert(s, Equals, targetSQLs[i][j])
 		}
 	}
 }
 
 func (s *testSyncerSuite) TestParseDDLSQL(c *C) {
+	// TODO: need refine cases
 	cases := []struct {
 		sql      string
 		schema   string
-		ignore   bool
 		isDDL    bool
```

Comment:
Should we remove `isDDL`? We don't check `isDDL` in `ParseDDLSQL` after this pr.